packages/@aws-cdk/aws-stepfunctions/lib/states/map.ts

```diff
@@ -198,12 +336,110 @@ export class Map extends State implements INextable {
     return errors;
   }
 
+  private validateProps(props: MapProps) {
+    if (props.itemSelector && props.parameters) {
+      throw new Error('Only one of itemSelector or parameters can be provided');
+    }
+    if (props.mode === MapProcessorMode.INLINE && props.distributatedMapOptions) {
+      throw new Error('distributatedMapOptions can only be used with DISTRIBUTED mode');
+    }
+    if (props.distributatedMapOptions?.label && props.distributatedMapOptions?.label.length > 40) {
+      throw new Error('Label can only be 40 characters long');
+    }
+    if (props.distributatedMapOptions?.itemBatcher) {
+      const itemBatcher = props.distributatedMapOptions?.itemBatcher;
+      if (itemBatcher.maxItemsPerBatch && itemBatcher.maxItemsPerBatchPath) {
+        throw new Error('Only one of maxItemsPerBatch or maxItemsPerBatchPath can be provided');
+      }
+      if (itemBatcher.maxInputBytesPerBatch && itemBatcher.maxInputBytesPerBatchPath) {
+        throw new Error('Only one of maxInputBytesPerBatch or maxInputBytesPerBatchPath can be provided');
+      }
+    }
+  }
```

Comment:
This long list of validations is an indication that we didn't get the UX right. There are so many ways a user can configure this wrong. We need to improve this by design.